### PR TITLE
feat: enable nvidia-device-plugin for Nvidia GPUs on AzurestackHub, and add unit test

### PR DESCRIPTION
**Pull request by penggu**
doc: add instruction for GPU cluster creation on Azure Stack Hub after fixing issue #4242.

**Reason for Change**:

Add documentation after fixing #4242 

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [x] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [x] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: docs/topics/gpu.md [hunk 1]
```diff
@@ -4,6 +4,18 @@ If you created a Kubernetes cluster with one or multiple Ubuntu node pools whose
 
 The NVIDIA drivers are automatically installed on every GPU agent in your cluster, so you don't need to do that manually, unless you require a specific version of the drivers. Currently, the installed driver is version 418.40.04.
 
+On Azure Stack Hub, however, NVIDIA drivers are not automatically installed. In this case the nvidia-device-plugin addon need to be explicitly enabled in apimodel.json, as follows:
```
Comment (jackfrancis) [inline docs/topics/gpu.md:7]: @jadarsie why are drivers not automatically installed for Azure Stack Hub?

Comment (jadarsie) [reply docs/topics/gpu.md:7]: Not aware, it should be based on my understanding, will investigate.

**Comment by jadarsie** [comment]
https://github.com/Azure/aks-engine/blob/30bb3381fdc7cdfee6ab734c1d0e9224d4dd509d/pkg/api/addons.go#L212

@penggu, based on the line above, it looks like enabling the addon explicitly wont work, did you try updating this line?

**Comment by jadarsie** [comment]
> updating

and by `updating` I mean removing the `!cs.Properties.IsAzureStackCloud()` condition

**Comment by penggu** [comment]
> https://github.com/Azure/aks-engine/blob/30bb3381fdc7cdfee6ab734c1d0e9224d4dd509d/pkg/api/addons.go#L212
> 
> @penggu, based on the line above, it looks like enabling the addon explicitly wont work, did you try updating this line?

I did not. I am now wondering why.

File: docs/topics/gpu.md (hunk 1 above)
Comment (jackfrancis) [reply docs/topics/gpu.md:7]: Looks like no one remembers why the nvidia driver functionality was explicitly disabled for Azure Stack?

Comment (jadarsie) [reply docs/topics/gpu.md]: GPUs are relatively new, that's why it was not allowed, but I thought the ban was already lifted.

**Comment by jadarsie** [comment]
the PR title needs a refresh

**Comment by penggu** [comment]
@jackfrancis I will update the unit test. Thanks for the help!

**Pull request closed without merging**